services/gundeck/src/Gundeck/Aws.hs

```diff
@@ -236,7 +236,11 @@ updateEndpoint :: Set UserId -> Token -> EndpointArn -> Amazon ()
 updateEndpoint us tk arn = do
   let req = over SNS.setEndpointAttributes_attributes fun (SNS.newSetEndpointAttributes (toText arn))
   env <- ask
-  res <- retrying (limitRetries 1) (const isTimeout) (const (sendCatch (env ^. awsEnv) req))
+  res <-
+    retrying
+      (exponentialBackoff 50000 <> limitRetries 1)
+      (const $ pure . or . flip map [isTimeout, isRateLimited] . (&))
```

Comment:
```suggestion
      (const $ pure . or . flip map [isTimeout, isRateLimited] . (&))
```
Isn't there a more easy way to express this?

Reply:
Also maybe a function can be abstracted for this to keep things more DRY

Reply:
`const $ \x -> isTimeout x || isRateLimited x` or `const $ liftM2 (||) isTimeout isRateLimited`.

Reply:
The first version looks easy to me

Reply:
Both done.